lib/driver.js

```diff
@@ -394,7 +394,7 @@ class XCUITestDriver extends BaseDriver {
       await this.wda.uninstall();
       this.logEvent('wdaUninstalled');
     } else if (!util.hasValue(this.wda.webDriverAgentUrl)) {
-      log.debug(`Capability 'useNewWDA' set to false, so trying to reuse currently running WDA instance at ${this.wda.url}...`);
+      log.debug(`Capability 'useNewWDA' set to false, so trying to reuse currently running WDA instance at ${this.wda.url.href}...`);
```

Comment:
there might be a bug with webDriverAgentUrl variable. inside WebDriverAgent class we just assign it as is, from options: `this.webDriverAgentUrl = args.webDriverAgentUrl;`, but in the code we assume it is an URL object and try to extract port and hostname attributes from it: 

```
    if (this.webDriverAgentUrl) {
      log.info(`Using provided WebdriverAgent at '${this.webDriverAgentUrl}'`);
      this.url = this.webDriverAgentUrl;
      this.setupProxies(sessionId);
      return this.webDriverAgentUrl;
    }

....

    const proxyOpts = {
      server: this.url.hostname,
      port: this.url.port,
      base: '',
      timeout: this.wdaConnectionTimeout,
    };

```